### PR TITLE
111 update api release guidelines

### LINKED ISSUE #93: Naming convention for Maintenance branches and branch protection rule for them
**Problem description**

Do we have already a naming convention for maintenance branches to enable patch release after the M4 release of a working group and/or API repository? The Guideline is currently only saying "the patched public API version x.y.z+1 shall be created through a maintenance release on a separate branch **referred to as a maintenance branch**"

In addition we need an additional branch protection rule in all repositories which will protect the maintenance branches like the `main` branch.

**Possible evolution**

A naming convention for maintenance branches is defined and added to https://github.com/camaraproject/ReleaseManagement/blob/main/documentation/API_Release_Guidelines.md.

A branch protection rule which covers these branch names  is rolled out across all repositories. 

Proposal:

`maintenance-rx`, created based on the release tag rx.n+1 (the M4 release) within the `main` branch. The first patch release on the maintenance-rx branch is then rx.n+2.

Exceptions for Commonalities and ICM within Fall24 Meta-Release: maintenance-r0.4 respective maintenance-r0.2 (but only for Fall24, proposal is that they follow in next release cycles also the rx.n format. Issue for that to be created).

Proposal for the branch protection rule is to extend the pattern for the current rule from `main` to `main*` which has the advantage that no new rule has to be created and for new repositories still only one rule need to created.  That would protect all branches which start with "main" - which shouldn't be a problem as branches for PR development should be in forks of the upstream repository not in the repository itself.

**Alternative solution**

**Additional context**

https://github.com/camaraproject/IdentityAndConsentManagement/pull/196 is waiting for a decision on this issue.

### LINKED ISSUE #101: Internal links in a release API spec
**Problem description**
There are some APIs that have images supporting the documentation, e.g. QoD or KYC, but those links currently always point to the main version, regardless their version. If we change the resource in a later release, the version in the previous release would show the new one, or if url changes, the reference would be broken.

**Possible evolution**
Links in releases should point to the resource packed in the same release

### LINKED ISSUE #111: Update API_Release_Guidelines.md with feedback on process
**Problem description**
Process description not fully aligned with the final process with PRs and PR reviews

**Expected action**
Update the description with the enhanced process

### PR DESCRIPTION
**Pull request by tanjadegroot**
#### What type of PR is this?

Add one of the following kinds:
* documentation

#### What this PR does / why we need it:

Update the API Release Guidelines based on feedback from the meta-release

#### Which issue(s) this PR fixes:

Fixes #111 
Fixes #93
Fixes #101  

#### Special notes for reviewers:

Corresponding updates have been made to the Wiki (mostly) but some more updates are needed there.

### CODE REVIEW

File: documentation/API_Release_Guidelines.md [hunk 1]
```diff
@@ -90,66 +94,75 @@ Example of continuous release numbering of an API version across its release typ
 | Release type | API version | release tag | release package | release package label |
 |------|------|:------:|:------:|:------:| 
 | N/A | work-in-progress | N/A | N/A | N/A |
-| pre-release | alpha | rx.1 ... rx.m | optional | optional: "pre-release" |
-| pre-release | release-candidate | rx.m+1 ... rx.n | mandatory | "pre-release" |
-| release | public | rx.n+1 | mandatory | "latest" |
-| maintenance release | public | rx.n+2 ... rx.n+p | mandatory | "latest" |
+| pre-release | alpha | rx.1 ... rx.m | optional | optional: `pre-release` |
```
Comment (hdamker) [inline documentation/API_Release_Guidelines.md:97]: I would delete "optional" in the last column of an alpha release ... the release is optional, but if done it is always a pre-release:

```suggestion
| pre-release | alpha | rx.1 ... rx.m | optional | `pre-release` |
```

File: documentation/API_Release_Guidelines.md [hunk 2]
```diff
@@ -90,66 +94,75 @@ Example of continuous release numbering of an API version across its release typ
...
+* Additional release-candidate API versions 1.1.0-rc.q may be released. For each such release-candidate API version, set the API version to `wip` in the first API update PR, and only set the next API version in the release PR of the next pre-release. The rc number evolves with each following pre-release numbered rx.m+1 - rx.n.
 * When the API version is ready for public release, create the release PR that sets the public API version to 1.1.0. (this PR minimally removes the rc extensions from the version and URL fields in the API yaml file and assures all API release assets are available as per the API readiness checklist).
-* After release PR approval, create the release rx.n+1 and update the API release tracker.
+* After release PR approval, create the release rx.n+1 for the public API version 1.1.0 and update the API release tracker.
 * The approved public API version 1.1.0 will be included in the meta-release.
+* If a patch is provided for the released API, it shall be declared asthe latest release, and the information shall be published on the existing API release tracker.
```
Comment (hdamker) [inline documentation/API_Release_Guidelines.md:168]:
```suggestion
* If a patch is provided for the released API, it shall be declared as the latest release, and the information shall be published on the existing API release tracker.
```

**Review by hdamker — COMMENTED** [review]
Some minor suggestions, looks overall good to me (have reviewed only the changes).

File: documentation/API_Release_Guidelines.md (hunk 1 above)
Comment (tanjadegroot) [reply documentation/API_Release_Guidelines.md:104]: makes sense, done

**Review by hdamker — APPROVED** [review]
LGTM

**Review by soadeyemo — APPROVED** [review]
Looks good.

**Pull request merged**